llama-index-core/llama_index/core/node_parser/text/semantic_splitter.py

```diff
@@ -227,13 +227,13 @@ def _build_node_chunks(
             start_index = 0
 
             for index in indices_above_threshold:
-                end_index = index - 1
+                end_index = index
```

Comment:
I believe this needs to be this way because the distances are computed between "i" and "i+1". So if the distance between i and i+1 is bigger than the threshold, we don't want them to be part of the same group. 

In particular if `index` is in `indices_above_threshold`, that means we want to separate out `index` and `index + 1`. So I believe from that we should be grouping together sentences in the slice `[start_index: index + 1]` as this will not go from `start_index ,...., index`.

Reply:
With the example provided by @hkristof03:

Before suggested fix: (only showing text of chunks in below snippets)
```
chunks = ["", "Hill testified against his former associates..."]
```

After suggested fix:
```
chunks = ["Hill testified against his former associates... ", "His testimony led to 50 convictions.  .."
```